### DISCUSSION TITLE
best way to schedule cli import

### DISCUSSION THREAD

**Discussion by ajhawley** [immich-app/immich#967] [Q&A] [answered]
i want to automate the cli to run every hour 

images will be added to the directory from my iphone using photosync app via samba as i dont want to reupload my entire album from my iphone 

i use this command but just not sure what is the best way to automate it

node opt/immich-cli/bin/index.js upload --yes --delete --email user@email.com --password immich1234 --server http://127.0.0.1:2283/api -d /home/user/immich-upload/

**Comment by bo0tzz** [comment, marked as answer]
The best tool is probably a [cron job](https://en.wikipedia.org/wiki/Cron). You'll need to google for the exact way to set it up on the operating system you're using.

**Reply by ajhawley** [reply]
i was thinking cron but wasn't sure entirely how to set it up, i will have to have a play about, im using ubuntu 22

**Reply by bertmelis** [reply]
I have a systemd timer that runs immich-cli in a container. I must admit I use Cockpit as server GUI which makes creating systemd timers easy.

**Reply by ajhawley** [reply]
ok so here is a bit of info on how i got it working just incase any other noob wants to automate the cli

cd /etc/
sudo nano crontab

add the command to the cron file, here is my example, it runs every 30 mins and checks for media in a directory i created called immich-upload 

*/30 * * * *     user    node /opt/immich-cli/bin/index.js upload --yes --delete --email user@email.com --password immichapp --server http://127.0.0.1:2283/api -d /home/user/immich-upload/ >/dev/null 2>&1

if you change the time from 30mins to 1 and put an image in the directory it should be uploaded to your immich directory after 1 minute if the cron ran without issues
